datadog_lambda/handler.py

```diff
@@ -15,6 +18,27 @@ class HandlerError(Exception):
     pass
 
 
+class _ErrorOutsideHandlerDecorator(object):
```

Comment:
Why a decorator and not just calling the function?

I'm trying to understand the reason behind it, but this just looks like a wrapper to the `emit_telemetry_on_exception_outside_of_handler` which then re-raises the error.

Reply:
Good question, the purpose of the decorator is to get the request ID into the trace (necessary for the frontend to display traces in the invocation table). If we just call `emit_telemetry_on_exception_outside_of_handler` and re-raise the exception on line 61, we won't have access to the lambda context which holds the request ID

Reply:
Ah, interesting, you are right. Thanks for explaining, that makes a lot of sense!

Reply:
Wait, maybe the naming is not ideal. This is not a decorator, this is a fallback-handler for this scenario. What do you think? Maybe I'm wrong, but we are not decorating any handler here, we are creating a new handler due to the customers code not being able to be imported.

Reply:
I agree, updated to `_ErrorFallbackHandler` instead, but open to other naming ideas if you've got something in mind